Scripts/upload_test_results.sh

```diff
@@ -0,0 +1,165 @@
+#!/bin/bash
+
+# Uploads a test results file from fastlane/test_output/sdk/**/*.junit to the test observability server.
+# Must be run from root of repo.
+
+# Options:
+# -u / --upload-server-base-url <url>: Allows you to specify a URL to use as the upload server base URL. Defaults to https://test-observability.herokuapp.com.
+# -i / --iteration <number>: If running the tests in a loop inside a single CI job, indicates which iteration of the loop is currently executing. Defaults to 1.
+
+set -e
+
+# 1. Check dependencies.
+
+if ! which jq > /dev/null
+then
+  echo "You need to install jq." 2>&1
+  exit 1
+fi
+
+# 2. Grab the variables from the environment.
+
+if [[ -z $TEST_OBSERVABILITY_SERVER_AUTH_KEY ]]
+then
+  echo "The TEST_OBSERVABILITY_SERVER_AUTH_KEY environment variable must be set." 2>&1
+  exit 1
+fi
+
+if [[ -z $GITHUB_REPOSITORY ]]
+then
+  echo "The GITHUB_REPOSITORY environment variable must be set." 2>&1
+  exit 1
+fi
+
+if [[ -z $GITHUB_SHA ]]
+then
+  echo "The GITHUB_SHA environment variable must be set." 2>&1
+  exit 1
+fi
+
+if [[ -z $GITHUB_REF_NAME ]]
+then
+  echo "The GITHUB_REF_NAME environment variable must be set." 2>&1
+  exit 1
+fi
+
+if [[ -z $GITHUB_RETENTION_DAYS ]]
+then
+  echo "The GITHUB_RETENTION_DAYS environment variable must be set." 2>&1
+  exit 1
+fi
+
+if [[ -z $GITHUB_ACTION ]]
+then
+  echo "The GITHUB_ACTION environment variable must be set." 2>&1
+  exit 1
+fi
+
+if [[ -z $GITHUB_RUN_NUMBER ]]
+then
+  echo "The GITHUB_RUN_NUMBER environment variable must be set." 2>&1
+  exit 1
+fi
+
+if [[ -z $GITHUB_RUN_ATTEMPT ]]
+then
+  echo "The GITHUB_RUN_ATTEMPT environment variable must be set." 2>&1
+  exit 1
+fi
+
+if [[ -z $GITHUB_RUN_ID ]]
+then
+  echo "The GITHUB_RUN_ID environment variable must be set." 2>&1
+  exit 1
+fi
+
+if [[ -z $GITHUB_BASE_REF ]]
+then
+  echo "The GITHUB_BASE_REF environment variable must be set." 2>&1
+  exit 1
+fi
+
+if [[ -z $GITHUB_HEAD_REF ]]
+then
+  echo "The GITHUB_HEAD_REF environment variable must be set." 2>&1
+  exit 1
+fi
+
+if [[ -z $GITHUB_JOB ]]
+then
+  echo "The GITHUB_JOB environment variable must be set." 2>&1
+  exit 1
+fi
+
+# 3. Grab command-line options.
+
+# https://stackoverflow.com/questions/192249/how-do-i-parse-command-line-arguments-in-bash
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        -i|--iteration) iteration="$2"; shift ;;
+        -u|--upload-server-base-url) upload_server_base_url="$2"; shift ;;
+        *) echo "Unknown parameter passed: $1"; exit 1 ;;
+    esac
+    shift
+done
+
+if [[ -z $iteration ]]
+then
+  iteration=1
+fi
+
+# 4. Find the JUnit test report.
+
+test_reports=$(find fastlane/test_output/sdk -name '*.junit')
```

Comment:
Is this path correct?
The path used before (for the Fastlane output) was `fastlane/test_output/examples/`
Moreover, maybe it's a better idea to pass this path as an argument to `upload_test_results.sh`?

Reply:
There are two paths that are used for Fastlane output:

- `fastlane/test_output/sdk/` is used for the test output from the SDK’s tests
- `fastlane/test_output/examples/` is used for the test output from the example app’s tests

In this script, we only care about the SDK’s test results.

Why do you think it might be better to pass the path as an argument?

Reply:
> Why do you think it might be better to pass the path as an argument?

I'm guessing we can configure it from within the Fastfile, hence if someone will change it there, they have to change it here too. But I just realized this doesn't solve it, maybe better idea is to set these paths as ENV var ... I just wonder ;)

Reply:
anyway I approved this pull request